src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -111,18 +112,30 @@
     Text getDescription();
 
     /**
-     * Gets all subjects that have this permission set.
+     * Gets all loaded subjects that have this permission set.
      *
      * <p>If you want to know to which role-templates this permission is
      * assigned use {@link PermissionService#SUBJECTS_ROLE_TEMPLATE}.
      *
      * @param type The subject type identifier to use
-     * @return An immutable set of subjects that have this permission set
+     * @return An immutable map of subjects that have this permission set
      * @see SubjectCollection#getAllWithPermission(String)
      */
     Map<Subject, Boolean> getAssignedSubjects(String type);
 
     /**
+     * Searches for and returns all subjects that have this permission set.
+     *
+     * <p>If you want to know to which role-templates this permission is
+     * assigned use {@link PermissionService#SUBJECTS_ROLE_TEMPLATE}.
+     *
+     * @param type The subject type identifier to use
+     * @return An immutable map of subjects that have this permission set
+     * @see SubjectCollection#getAllWithPermission(String)
+     */
+    CompletableFuture<Map<SubjectReference, Boolean>> findAssignedSubjects(String type);
```

Comment:
This isn't the goal of this PR but should you be willing to do further permissions work @lucko then maybe look into changing this parameter to take something akin to a catalog? Having arbitrary string values doesn't make for good API as without reading the docs I'd have no idea what to pass in here.